build.gradle

```
@@ -215,7 +209,7 @@ task integrationTest(type: Test) {
     testClassesDirs = sourceSets.integrationTest.output.classesDirs
     classpath = sourceSets.integrationTest.runtimeClasspath
 
-    dependsOn test, distTar
+    dependsOn testClasses, distTar
```

I'm probably lacking some context, but why would we need to avoid running test before integration tests? 
If we really want to, we could use `./gradlew integrationTest -x test`.

Also, would like to understand more the dependency to distTar for integration tests.

First time configuring IT on gradle, so looking at https://docs.gradle.org/7.3/userguide/java_testing.html#sec:configuring_java_integration_tests
`shouldRunAfter test` may be a more idiomatic way of defining this? 

Also, we could use `check.dependsOn integrationTest` to link ITs to gradle check, but we can do that on a separate PR to tackle #221

---

Requiring people to explicitly opt out of running unit tests before running integration tests seems less friendly and was a bit of a headache personally when writing new integration tests for https://github.com/aiven/jdbc-connector-for-apache-kafka/pull/223. This is similar to how Kafka does it, although they go one step further and define three different Gradle goals: `unitTest`, `integrationTest`, and `test` (which runs both unit and integration tests).

RE the dependency on `distTar`: I'm honestly not sure about this one, but chose to leave it in for now as it doesn't appear to be causing any issues.

---

Got it, that make sense. Then, for connectors: `test` is for unit-tests, and `integrationTest` for ITs and runs independently from unit-tests.

May be worth adding a note to clarify this?
```suggestion
    // ITs run independent from unit-tests
    dependsOn testClasses, distTar
```
 
Also, could `gradle check` be the way to both tests, or `test integrationTest` should suffice?

---

👍 for adding the comment

I tried attaching the `integrationTest` task to `check` but the end result was that `./gradlew build -x test` ended up running integration tests (since we didn't exclude that task explicitly). So at least with the latest version of the PR, you have to run `test integrationTest` instead of just `check` if you'd like to run both categories of tests.

---

oh I see. That make sense. Thanks @C0urante